### PR TITLE
fix(NIPT orders get the wrong priority)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

As raised by @vwirta in this issue: https://github.com/Clinical-Genomics/cg/issues/1474 
NIPT orders are Pools and automatically are assigned research priority on invoices.

NIPT orders are only made by cust032

### Added

- In the check whether an order is a pool, added a clause that it should excluce cust032 from this check
- In testing store_helpers.py added several helpers: one adds an invoice the other a pool to store

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b NIPT_priority -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/invoice.py [hunk 1]
```diff
@@ -93,7 +93,11 @@ def prepare(self, costcenter: str) -> dict:
     def _discount_price(self, record, discount: int = 0):
         """Get discount price for a sample or pool."""
 
-        if self.record_type == "Pool" or record.priority_human == "clinical trials":
+        if (
+            self.record_type == "Pool"
+            and not self.customer_obj.internal_id == "cust032"
+            or record.priority_human == "clinical trials"
```
Comment (henrikstranneheim) [inline cg/meta/invoice.py:99]: I think all these hard coded strs should be constants (if they do not already exist)

File: cg/meta/invoice.py [hunk 2]
```diff
@@ -146,7 +150,11 @@ def prepare_record(self, costcenter: str, discount: int, record):
         order = record.order
         ticket = record.ticket if self.record_type == "Pool" else record.original_ticket
         lims_id = None if self.record_type == "Pool" else record.internal_id
-        priority = "research" if self.record_type == "Pool" else record.priority_human
+        priority = (
+            "research"
```
Comment (henrikstranneheim) [inline cg/meta/invoice.py:154]: Same here

File: tests/meta/test_invoice.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from tests.store_helpers import StoreHelpers
+from cg.meta.invoice import InvoiceAPI
+from cg.apps.lims import api as limsAPI
+import cg.store as Store
+from cg.store import models
+
+
+def test_invoice_api(store: Store, lims_api: limsAPI, helpers: StoreHelpers):
+    # GIVEN a invoice
```
Comment (henrikstranneheim) [inline tests/meta/test_invoice.py:9]:
```suggestion
    # GIVEN an invoice
```

File: tests/meta/test_invoice.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+from tests.store_helpers import StoreHelpers
+from cg.meta.invoice import InvoiceAPI
+from cg.apps.lims import api as limsAPI
+import cg.store as Store
+from cg.store import models
+
+
+def test_invoice_api(store: Store, lims_api: limsAPI, helpers: StoreHelpers):
```
Comment (henrikstranneheim) [inline tests/meta/test_invoice.py:8]: Add doc str

File: tests/meta/test_invoice.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    # GIVEN a invoice
+    invoice = helpers.ensure_invoice(
+        store,
+        invoice_id=0,
+        record_type="Sample",
```
Comment (henrikstranneheim) [inline tests/meta/test_invoice.py:13]: Use constant

File: tests/meta/test_invoice.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    )
+
+    # THEN calling InvoiceAPI should return an API
+    api = InvoiceAPI(store, lims_api, invoice)
+    assert api
+    # THEN record_type should be Sample
+    assert api.record_type == "Sample"
+    # THEN prepare should return customer information with customer id cust999
+
+    kth_inv = api.prepare("ki")
```
Comment (henrikstranneheim) [inline tests/meta/test_invoice.py:24]:
```suggestion
    # THEN prepare should return customer information with customer id cust999
    kth_inv = api.prepare("ki")
```
Feels like the comment needs an update

File: tests/store_helpers.py [hunk 7]
```diff
@@ -1,643 +1,741 @@
-"""Utility functions to simply add test data in a cg store"""
```
Comment (henrikstranneheim) [inline tests/store_helpers.py:1]: A black issue or what happened here?

File: tests/test_store_helpers.py [hunk 8]
```diff
@@ -24,3 +24,37 @@ def test_add_microbial_sample(base_store, helpers):
     assert sample.customer
     # THEN there should be an application_version
     assert sample.application_version
+
+
+def test_add_invoice(base_store, helpers):
```
Comment (henrikstranneheim) [inline tests/test_store_helpers.py:29]: Add type hints

File: tests/test_store_helpers.py [hunk 9]
```diff
@@ -24,3 +24,37 @@ def test_add_microbial_sample(base_store, helpers):
...
+    # GIVEN a base_store
+
+    # WHEN using the helper to add an invoice with a sample
+    invoice_sample = helpers.add_invoice(base_store, type="Sample", id=1)
+
+    # THEN there should be an invoice
+    assert invoice_sample
+
+    # THEN there should be
```
Comment (henrikstranneheim) [inline tests/test_store_helpers.py:38]:
```suggestion
    # THEN there should be a sample
```

File: tests/test_store_helpers.py [hunk 10]
```diff
@@ -24,3 +24,37 @@ def test_add_microbial_sample(base_store, helpers):
...
+    assert invoice_sample.sample
+
+    # THEN there should be a customer
+    assert invoice_sample.customer
+
+    # THEN customer should have an id
+
+    assert invoice_sample.customer_id
```
Comment (henrikstranneheim) [inline tests/test_store_helpers.py:46]:
```suggestion
    # THEN there should be a customer id
    assert invoice_sample.customer_id
```

File: tests/test_store_helpers.py [hunk 11]
```diff
@@ -24,3 +24,37 @@ def test_add_microbial_sample(base_store, helpers):
...
+
+    # THEN there should not be a pool
+    assert not invoice_sample.pool
+    # WHEN using the helper to add an invoice with a pool
+    invoice_pool = helpers.add_invoice(base_store, type="Pool", id=2)
```
Comment (henrikstranneheim) [inline tests/test_store_helpers.py:51]: Use constant for "Pool"

**Review by henrikstranneheim — APPROVED** [review]
Nice! Great that you added tests ⭐

**Comment by ChrOertlin** [comment]
Closed this branch due to that capital letters prevented deployment on cg-stage-vm1. New pull request created https://github.com/Clinical-Genomics/cg/pull/1746

**Pull request closed without merging**